joern-cli/frontends/jimple2cpg/src/main/scala/io/joern/jimple2cpg/Jimple2Cpg.scala

```diff
@@ -75,101 +52,77 @@ class Jimple2Cpg extends X2CpgFrontend[Config] {
     Options.v().set_src_prec(Options.src_prec_class)
   }
 
-  def sootLoadSource(input: String, ext: String): Unit = {
-    // Soot does not support loading single class/jimple file using path, so we move it to temp dir first
-    // NOTE: Soot’s frontend for Java source files is outdated (only partially supports Java version up to 7) and not very robust.
-    val src = new JFile(input)
-    val dst = new JFile(ProgramHandlingUtil.getUnpackingDir.toString, src.getName)
-    val prec = ext match {
-      case "jimple" => Options.src_prec_jimple
-      case _        => Options.src_prec_class
+  /** Load all class files from archives or directories recursively
+    * @return
+    *   The list of extracted class files and the components of their fully qualified class names
+    */
+  private def loadClassFiles(src: File, tmpDir: File): List[ClassFile] = {
+    val archiveFileExtensions = Set(".jar", ".war", ".zip")
+    extractClassesInPackageLayout(
+      src,
+      tmpDir,
+      isClass = e => e.extension.contains(".class"),
```

Comment:
You can use `extension.exists(_ == ".class")` which should take advantage of the equality check's performance. Strings are check by value in Scala.